analyzers/tests/SonarAnalyzer.UnitTest/MetadataReferences/NuGetMetadataFactory.cs

```diff
@@ -66,67 +64,36 @@ public static partial class NuGetMetadataFactory
         public static IEnumerable<MetadataReference> Create(string packageId, string packageVersion, string runtime = null) =>
             Create(new Package(packageId, packageVersion, runtime), SortedAllowedDirectories);
 
-        public static IEnumerable<MetadataReference> CreateNetStandard21()
-        {
-            var packageDir = Path.Combine(PackagesFolder, @"NETStandard.Library.Ref.2.1.0\ref\netstandard2.1");
-            if (Directory.Exists(packageDir))
-            {
-                LogMessage($"Package found at {packageDir}");
-            }
-            else
-            {
-                LogMessage($"Package not found at {packageDir}");
-                PackageManager.InstallPackage("NETStandard.Library.Ref", SemanticVersion.ParseOptionalVersion("2.1.0"), ignoreDependencies: true, allowPrereleaseVersions: false);
-                if (!Directory.Exists(packageDir))
-                {
-                    throw new ApplicationException($"Test setup error: folder for downloaded package does not exist. Folder: {packageDir}");
-                }
-            }
-
-            return Directory.GetFiles(packageDir, "*.dll", SearchOption.AllDirectories)
-               .Select(x => (MetadataReference)MetadataReference.CreateFromFile(x))
-               .ToImmutableArray();
-        }
-
         /// <param name="allowedDirectories">List of allowed directories sorted by preference to search for DLL files.</param>
         private static IEnumerable<MetadataReference> Create(Package package, string[] allowedDirectories)
         {
-            Console.WriteLine();
-            Console.WriteLine($"Package: {package.Id}, {package.Version}");
             package.EnsurePackageIsInstalled();
-
-            var packageDirectory = package.PackageDirectory();
-            if (!Directory.Exists(packageDirectory))
+            var dllsPerDirectory = new Dictionary<string, IEnumerable<string>>();
+            var packageLibDirectory = package.PackageDirectory();
+            var foldersWithDlls = new List<string> { Path.Combine(packageLibDirectory, "lib"), Path.Combine(packageLibDirectory, "ref") };
+            foreach (var folder in foldersWithDlls.Where(Directory.Exists))
```

Comment:
We should not not need the list, we won't add to it.
```suggestion
            foreach (var folder in new[] { "lib", "ref" }.Select(x => Path.Combine(packageLibDirectory, x)).Where(Directory.Exists))

```